google_guest_agent/network/systemd_networkd.go

```diff
@@ -0,0 +1,112 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+const configDir = "/etc/systemd/network"
+
+type systemdnetworkd struct{}
+
+func (n systemdnetworkd) IsManaging(ctx context.Context, iface string) (bool, error) {
+	// First check if the service is running.
+	res := run.WithOutput(ctx, "/bin/sh", "-c", "systemctl status systemd-networkd.service | grep Active:")
```

Comment:
Not the most effective way of checking if a unit is active, please consider doing the following:

```
  res := run.WithOutput(ctx, "/bin/sh", "-c", "systemctl is-active systemd-networkd.service")
  if res.ExitCode != 0 {
    return false, nil
  }
```

If the unit is active it will return a 0 exit code, it will return exit code 1 otherwise.